src/analysis/Environments.test.ts

```diff
@@ -159,7 +159,7 @@ describe('Environments', () => {
       asMockedFn(listRuntimesV2).mockReturnValue(Promise.resolve([runtime1]));
 
       await act(async () => {
-        await render(h(Environments, { nav: defaultNav }));
+        render(h(Environments, { nav: defaultNav }));
       });
 
       const tableRows: HTMLElement[] = screen.getAllByRole('row').slice(1); // skip header row
```

Comment:
Would it be better to instead rewrite these tests so they doesn't use `act()`? It looks like you could instead wait for the spinner to disappear, or for the element the test needs to appear. This seems to work:
```
      render(h(Environments, { nav: defaultNav }));
      const rows = await screen.findAllByRole('row');

      const tableRows: HTMLElement[] = rows.slice(1); // skip header row
```

This approach makes more sense to me but maybe I'm missing a reason why `act()` would be better here.

Reply:
That's one of the reasons for pushing custom hooks to encapsulate data fetching. With custom hooks, we can mock the return value of the hook in tests and avoid async operations in tests.

Reply:
So I guess to answer your question... yes, but that would be a larger project that would affect app code as well as test code.

Reply:
Maybe I'm missing something, but I've rewritten the tests in this way (without `act`) and they still work. I understand that `act` is used to handle async data fetching, but my understanding is that `render` is already calling `act` internally. And since the tests work without `act`, I'm wondering why it's needed.

Reply:
If I remove the `act` from around `render` in Environments.test, the test fails.